### PR TITLE
refactor: simplify round function and make it faster

### PR DESCRIPTION
**Pull request by tarikjabiri**

### CODE REVIEW

**Comment by tarikjabiri** [comment]
@microsoft-github-policy-service agree

**Comment by danmarshall** [comment]
Hi @tarikjabiri , thanks so much for this valuable contribution! This is definitely one of the most significant bottlenecks in the code. (see #412 ). It is similar to the [original code](https://github.com/microsoft/maker.js/pull/254/commits/1a2a4dfe7da22f139b8b76e34e82f70320415155) but I didn't use Epsilon for ES5 compatibility.

I'm not sure if people still need support for ES5. I will keep this PR open while I investigate.
I probably should have used a polyfill for epsilon at the time, but Maker.js started with an early TypeScript version and has an antique build system by today's standards.

Note, [here is the reference code on MDN from the Wayback machine](http://web.archive.org/web/20170505224314/https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/round) which has been removed in the [current document](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/EPSILON).

**Comment by danmarshall** [comment]
Instead of adding a polyfill to `Number`, I've arrived at defining a scoped variable EPSILON, which can be `Number.EPSILON || Math.pow(2, -52)`. Would you be willing to add this in?

**Comment by tarikjabiri** [comment]
Thats a good idea, done

File: packages/maker.js/src/core/maker.ts [hunk 1]
```diff
@@ -130,8 +130,9 @@ namespace MakerJs {
         if (n % 1 === 0) return n;
 
         const temp = 1 / accuracy;
+        const eps = Number.EPSILON || Math.pow(2, -52);
```
Comment (danmarshall) [inline packages/maker.js/src/core/maker.ts:133]: can we move this to the outer scope?

Comment (tarikjabiri) [reply packages/maker.js/src/core/maker.ts:133]: As an export ?

Comment (danmarshall) [reply packages/maker.js/src/core/maker.ts:133]: no export yet. I imagine it may be promoted when needed though.

Comment (danmarshall) [reply packages/maker.js/src/core/maker.ts]: It will be in the scope of the namespace. So it won't evaluate the conditional `||` every call.

Comment (tarikjabiri) [reply packages/maker.js/src/core/maker.ts]: Done

**Comment by danmarshall** [comment]
Perfect, thank you!!

**Pull request merged**